### PR TITLE
Merge Save and Return branch

### PR DESCRIPTION
**Pull request by SandipKhedekar**

### CODE REVIEW

**Review by ElliotAndersonHMRC — DISMISSED** [review]
Looks good

File: app/views/annualallowance/taxyear/DefinedContributionAmountView.scala.html [hunk 1]
```diff
@@ -51,13 +52,13 @@
...
         .withPrefix(PrefixOrSuffix(content = Text("£")))
         .withHint(HintViewModel(HtmlContent(contentHint))))
 
-        @govukButton(
-            ButtonViewModel(messages("site.continue"))
-        )
+
+        @continueButton()
     }
 }
```
Comment (bharti5verma) [inline app/views/annualallowance/taxyear/DefinedContributionAmountView.scala.html:64]: End of Line missing

File: app/views/annualallowance/taxyear/PensionSchemeInput2016preAmountsView.scala.html [hunk 2]
```diff
@@ -73,8 +74,6 @@ <h1 class="govuk-heading-xl"> @messages("pensionSchemeInputAmounts.heading.2016-
             .withWidth(Fixed10)
         )
 
-        @govukButton(
-            ButtonViewModel(messages("site.continue"))
-        )
+        @continueButton()
     }
 }
```
Comment (bharti5verma) [inline app/views/annualallowance/taxyear/PensionSchemeInput2016preAmountsView.scala.html:79]: End of line missing

File: app/views/components/ContinueButton.scala.html [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+@if(frontendAppConfig.optionalAuthEnabled && request.signedIn) {
+    @govukButton(
+        ButtonViewModel(messages("site.saveAndContinue"))
+    )
+
+} else {
+    @govukButton(
+        ButtonViewModel(messages("site.continue"))
+    )
+}
```
Comment (bharti5verma) [inline app/views/components/ContinueButton.scala.html:35]: End of line missing

File: app/views/components/ContinueButtonWithLink.scala.html [hunk 4]
```diff
@@ -0,0 +1,38 @@
...
+
+        .asLink(continueURL.url)
+    )
+
+} else {
+    @govukButton(
+    ButtonViewModel(messages("site.continue"))
+    .asLink(continueURL.url)
+    )
+}
```
Comment (bharti5verma) [inline app/views/components/ContinueButtonWithLink.scala.html:38]: End of line missing

File: conf/app.routes [hunk 5]
```diff
@@ -396,3 +398,7 @@ GET        /annual-allowance/2016flexi-pre-pension-input-amount-defined-contribu
 POST       /annual-allowance/2016flexi-pre-pension-input-amount-defined-contribution                  controllers.annualallowance.taxyear.DefinedContribution2016PreFlexiAmountController.onSubmit(mode: Mode = NormalMode)
 GET        /annual-allowance/change-2016flexi-pre-pension-input-amount-defined-contribution                        controllers.annualallowance.taxyear.DefinedContribution2016PreFlexiAmountController.onPageLoad(mode: Mode = CheckMode)
 POST       /annual-allowance/change-2016flexi-pre-pension-input-amount-defined-contribution                        controllers.annualallowance.taxyear.DefinedContribution2016PreFlexiAmountController.onSubmit(mode: Mode = CheckMode)
+
+GET        /previous-claim-continue                        controllers.PreviousClaimContinueController.onPageLoad()
+POST       /previous-claim-continue                        controllers.PreviousClaimContinueController.onSubmit()
+GET        /previous-claim-navigate                         controllers.PreviousClaimContinueController.redirect()
```
Comment (bharti5verma) [inline conf/app.routes:404]: End of line missing

File: conf/messages.cy [hunk 6]
```diff
@@ -1048,83 +1051,93 @@ upliftFailed.contact = cy: You can contact us to access our services and speak t
...
+
+previousClaimContinue.title = cy: Do you want to continue with your previous claim?
+previousClaimContinue.heading = cy: Do you want to continue with your previous claim?
+previousClaimContinue.body = cy: If you start your claim from the beginning, your current claim will be removed.
+previousClaimContinue.hint = cy: Select no if you would like to start your claim from the beginning.
+previousClaimContinue.option1 = cy: Yes
+previousClaimContinue.option2 = cy: No
+previousClaimContinue.checkYourAnswersLabel = cy: Do you want to start your claim from the beginning?
+previousClaimContinue.error.required = cy: Select yes if you want to continue with your previous claim
+previousClaimContinue.change.hidden = cy: if you want to continue with your previous claim
```
Comment (bharti5verma) [inline conf/messages.cy:1143]: End of line missing

File: conf/messages.en [hunk 7]
```diff
@@ -1127,4 +1130,14 @@ definedContribution2016PostFlexiAmount.change.hidden = your pension input amount
...
+
+previousClaimContinue.title = Do you want to continue with your previous claim?
+previousClaimContinue.heading = Do you want to continue with your previous claim?
+previousClaimContinue.body = If you start your claim from the beginning, your current claim will be removed.
+previousClaimContinue.hint = Select no if you would like to start your claim from the beginning.
+previousClaimContinue.option1 = Yes
+previousClaimContinue.option2 = No
+previousClaimContinue.checkYourAnswersLabel = Do you want to start your claim from the beginning?
+previousClaimContinue.error.required = Select yes if you want to continue with your previous claim
+previousClaimContinue.change.hidden = if you want to continue your previous claim
```
Comment (bharti5verma) [inline conf/messages.en:1143]: End of line missing

File: test/controllers/setupquestions/ResubmittingAdjustmentControllerSpec.scala [hunk 8]
```diff
@@ -109,9 +110,9 @@ class ResubmittingAdjustmentControllerSpec extends SpecBase with MockitoSugar {
 
         status(result) mustEqual SEE_OTHER
 
-        val capturedUserAnswers = userAnswersCaptor.getValue
-        capturedUserAnswers.get(ResubmittingAdjustmentPage).get mustBe true
-        capturedUserAnswers.get(SavingsStatementPage(true)).get mustBe true
+//        val capturedUserAnswers = userAnswersCaptor.getValue
```
Comment (bharti5verma) [inline test/controllers/setupquestions/ResubmittingAdjustmentControllerSpec.scala:113]: Commented?

**Review by bharti5verma — APPROVED** [review]

**Pull request merged**